### ISSUE TITLE
[Feature Request]: a11y in Accordion component - change accordion button into heading

### ISSUE THREAD

**Issue by awojciak** [carbon-design-system/carbon#18823]
### The problem

In one of the pull requests in the project I'm working on I've got info that Accordion component should be ensured to comply to WCAG accessibility standards bychanging accordion buttons in Accordion component into headings, like in example here: https://www.w3.org/WAI/ARIA/apg/patterns/accordion/examples/accordion/

![Image](https://github.com/user-attachments/assets/c9422e58-e516-4aa1-b5ab-1f144e77da67)

### The solution

As I said in the problem description, accordion buttons should be wrapped in the headings, code of the example from WCAG page: https://codepen.io/pen

### Package

@carbon/react

### Business priority

Low Priority = release date is not dependent on fix or not upcoming

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)

**Comment by tay1orjones** [comment]
For `@carbon/react` we should be able to use the `Heading` component. This will make sure that the accordion uses the appropriate h tag in proper order based on nesting in the DOM.

We'll just need to make sure there's no styling conflicts and that there aren't any visual defects from the tag change (h1 through h6).
